### PR TITLE
Add config so we can override/initialize artsy_api_url

### PR DESCRIPTION
**Pull request by ashkan18**
# Problem
Reading from `ENV` to get `ARTSY_API_URL` or `gravity_url` can cause issue since the app that's going to use this gem might have used a different `ENV` for pointing to Artsy's api url. Ideally we can configure this gem in initializer of the app.

# Solution
Added `OmniAuth::Artsy.config` which currently initializes `artsy_api_url` using `ARTSY_API_URL` or `gravity_url` of ENV so it's backward compatible and app already using it are fine, also we can set it if we wanted to by

```ruby
# config/initializers/config.rb
OmniAuth::Artsy.configure do |config|
  config.artsy_api_url = ENV['ARTSY_API_URL']
end
```

Then we can use this from [`artsy-auth`](https://github.com/artsy/artsy-auth) and initialize everything from one place.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 0.2.1 (Next)
 ============
 
-* Your contribution here.
```
Comment (dblock) [inline CHANGELOG.md]: You probably want to put this back.

**Comment by dblock** [comment]
Generally the idiomatic Ruby  model for these that is also testable looks more like https://github.com/slack-ruby/slack-ruby-client/blob/master/lib/slack/config.rb, where you yield the configuration.

File: spec/lib/config_spec.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+require 'spec_helper'
+
+describe OmniAuth::Artsy::Config do
+  describe '#configure' do
+    before do
```
Comment (dblock) [inline spec/lib/config_spec.rb]: You should have an `after` with a `reset`.

**Comment by ashkan18** [comment]
Rubocop was failing in first few builds, fixed it by removing hardcoded rubocop version, upgrade fix it so it can run now, but doesn't like few things, going to fix them.

**Comment by dblock** [comment]
Lock rubocop version at latest, then run `rubocop -a ; rubocop --auto-gen-config` and you're good to go.

**Comment by ashkan18** [comment]
all right, that was interesting! should be all set now, rubocop was happy at somepoint but not in JRuby so updated that too and we should be good to go now.

**Pull request merged**